lib/datadog/appsec/response.rb

```diff
@@ -43,20 +43,22 @@ def negotiate(env)
         private
 
         FORMAT_MAP = {
-          'text/plain' => :text,
-          'text/html' => :html,
           'application/json' => :json,
+          'text/html' => :html,
+          'text/plain' => :text,
         }.freeze
 
-        DEFAULT_CONTENT_TYPE = 'text/plain'
+        DEFAULT_CONTENT_TYPE = 'application/json'
 
         def content_type(env)
           return DEFAULT_CONTENT_TYPE unless env.key?('HTTP_ACCEPT')
 
-          accepted = env['HTTP_ACCEPT'].split(',').map { |m| Utils::HTTP::MediaRange.new(m) }.sort!.reverse!
+          accept_types = env['HTTP_ACCEPT'].split(',').map(&:strip)
+
+          accepted = accept_types.map { |m| Utils::HTTP::MediaRange.new(m) }.sort!.reverse!
 
-          accepted.each_with_object(DEFAULT_CONTENT_TYPE) do |range, _default|
-            match = FORMAT_MAP.keys.find { |type| range === type }
+          accepted.each do |range|
```

Comment:
`accepted.each_with_object(DEFAULT_CONTENT_TYPE)` was making it so it returns `DEFAULT_CONTENT_TYPE` when no match has occured.

`accepted.each` now instead returns `accepted`, which is an array of `MediaRange`, so:

- the return value of `content_type` is now inconsistent: when a match is found it returns a content type but when none is found it returns an array of `MediaRange`
- when there is no match it does not return the default value anymore, which will break `FORMAT_MAP[content_type]`